### DISCUSSION TITLE
Recognize implicit this argument in C++ mangled names

### DISCUSSION THREAD

**Discussion by MrQubo** [Vector35/binaryninja-api#4796] [Analysis] [answered]
Apparently it was implemented some time ago: https://github.com/Vector35/binaryninja-api/issues/604
But I have never seen it working correctly. Do I need to enable it somehow?

My problem is that whenever there is a function with a mangled name, e.g. `void A::foo(int)` binaryninja sets the function type to `void ()(int)`, while the correct type would be `void ()(A *this, int)`.
It seems to even override normal analysis results, because usually I can clearly see an used, but uninitialized, register (the one for the second argument).

I'm working with 32-bit arm if that could be causing the problem.

**Comment by plafosse** [comment, marked as answer]
So 604 is different than this issue. To work around the issue you're seeing you need to uncheck the `Extract Types from Mangled Names` which we should probably disable by default.

<img width="784" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/10556585/b57a7d38-df86-4eff-a758-e14a385cb326">

**Reply by MrQubo** [reply]
Thanks. That does indeed work. Still, this feature (Extract Types From Mangled Names) is quite useful. It would be nice to see some improvements for it.
